From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Sat, 21 Jun 2014 16:50:18 +0200
Subject: [122/249] staging: rtl8723au: issue_probersp(): Don't insert beacon
 data twice
Origin: https://git.kernel.org/linus/d642a0a0b91d220f3c44bba84e54f8983b781fde

In case of AP mode, we were inserting the beacon data twice in the
probe_resp frame, leading to a corrupted on the wire frame.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/core/rtw_mlme_ext.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/rtl8723au/core/rtw_mlme_ext.c b/drivers/staging/rtl8723au/core/rtw_mlme_ext.c
index d1c40ae..4f9e5a9 100644
--- a/drivers/staging/rtl8723au/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8723au/core/rtw_mlme_ext.c
@@ -2592,7 +2592,8 @@ static void issue_probersp(struct rtw_adapter *padapter, unsigned char *da,
 						  cur_network->IELength -
 						  _FIXED_IE_LENGTH_);
 
-		memcpy(pframe, cur_network->IEs, cur_network->IELength);
+		memcpy(pframe, cur_network->IEs + _FIXED_IE_LENGTH_,
+		       cur_network->IELength - _FIXED_IE_LENGTH_);
 		pframe += cur_network->IELength;
 		pattrib->pktlen += cur_network->IELength;
 
